docs/TSG.md

```diff
@@ -140,6 +141,48 @@ From here, we simply went backwards from the exit event to find any errors; and
 
 This clearly shows that listener `7f30ac0dcff0` failed to register with the binding (i.e. UDP socket abstraction) because listener `7f30ac076d90` was already registered for the same ALPN. MsQuic only allows a single listener to be registered for a given ALPN on a local IP address and port.
 
+## Why is the API hanging or deadlocking?
+
+First, a bit of background. The MsQuic API has two types of APIs:
+
+- **Blocking / Synchronous** - These APIs run to completion and only return once finished. When running in the Windows kernel, these **MUST NOT** be called at `DISPATCH_LEVEL`. They are generally denoted by the `_IRQL_requires_max_(PASSIVE_LEVEL)` annotation. For example, [ConnectionClose](./api/ConnectionClose.md).
+- **Nonblocking / Asynchronous** - These APIs mearly queue work and return immediately. When running in the Windows kernel, these may be called at `DISPATCH_LEVEL`. They are generally denoted by the `_IRQL_requires_max_(DISPATCH_LEVEL)` annotation. For example, [StreamSend][./api/StreamSend.md].
```

Comment:
It might be worth it to do a quick run through the API and be able to remove "generally" from both of these statements. It would be nice to have the API's be documented that way if possible.

Reply:
Fixed and updated some annotations to be more permissive.